libs/secure-sm/src/main/java/org/elasticsearch/secure_sm/SecureSM.java

```diff
@@ -163,15 +163,18 @@ protected void checkThreadAccess(Thread t) {
         Objects.requireNonNull(t);
 
         // first, check if we can modify threads at all.
-        checkPermission(MODIFY_THREAD_PERMISSION);
+        boolean isInnocuousThread = isInnocuousThread(t);
+        if (isInnocuousThread == false) {
+            checkPermission(MODIFY_THREAD_PERMISSION);
```

Comment:
It's quite annoying that we need to put this check here, but I agree that it is appropriate, given the recent changes in JDK 19.